src/scalar.jl

```diff
@@ -52,7 +53,144 @@ transform(::Identity, x::Real) = x
+
+"""
+$(TYPEDEF)
+
+Scale transformation `x ↦ scale * x`.
+"""
+struct TVScale{T} <: ScalarTransform
+    scale::T
+    function TVScale{T}(scale::T) where {T}
+        @argcheck scale > zero(scale) DomainError
+        new(scale)
+    end
+end
+TVScale(scale::T) where {T} = TVScale{T}(scale)
+
+transform(t::TVScale, x::Real) = t.scale * x
+transform_and_logjac(t::TVScale{T}, x::Real) where {T<:Real} = transform(t, x), log(t.scale) 
+
+inverse(t::TVScale, x::Number) = x / t.scale
+inverse_and_logjac(t::TVScale{T}, x::Number) where {T<:Real} = inverse(t, x), -log(t.scale)
+
+"""
+$(TYPEDEF)
+
+Negative transformation `x ↦ -x`.
+"""
+struct TVNeg <: ScalarTransform
+end
+
+transform(::TVNeg, x::Real) = -x
+transform_and_logjac(t::TVNeg, x::Real) = transform(t, x), zero(x)
+
+inverse(::TVNeg, x::Number) = -x
+inverse_and_logjac(::TVNeg, x::Number) = -x, zero(x)
+
+####
+#### composite scalar transforms
+####
+"""
+$(TYPEDEF)
+
+A composite scalar transformation, i.e. a sequence of scalar transformations.
+The component transforms can be accessed indexing into the `CompositeScalarTransform`, 
+as in `ct[i]` or `ct[end]`.
+"""
+struct CompositeScalarTransform{Ts <: Tuple} <: ScalarTransform
+    transforms::Ts
+    function CompositeScalarTransform(transforms::Ts) where {Ts <: Tuple}
+        @argcheck length(transforms) > 0 DomainError
+        @argcheck all(t -> t isa ScalarTransform, transforms) DomainError
+        new{Ts}(transforms)
+    end
+end
+
+Base.getindex(t::CompositeScalarTransform, i) = t.transforms[i]
+Base.firstindex(t::CompositeScalarTransform) = firstindex(t.transforms)
+Base.lastindex(t::CompositeScalarTransform) = lastindex(t.transforms)
+
+transform(t::CompositeScalarTransform, x) = foldr((t, y) -> transform(t, y), t.transforms, init=x)
+function transform_and_logjac(ts::CompositeScalarTransform, x) 
+    foldr(ts.transforms, init=(x, zero(x))) do t, (x, logjac)
+        nx, nlogjac = transform_and_logjac(t, x)
+        x = nx
+        logjac += nlogjac
+        (x, logjac)
+    end
+end
+
+inverse(ts::CompositeScalarTransform, x) = foldl((y, t) -> inverse(t, y), ts.transforms, init=x)
+function inverse_and_logjac(ts::CompositeScalarTransform, x) 
+    foldl(ts.transforms, init=(x, zero(x))) do (x, logjac), t
+        nx, nlogjac = inverse_and_logjac(t, x)
+        x = nx
+        logjac += nlogjac
+        (x, logjac)
+    end
+end
+
+Base.:∘(t::ScalarTransform, s::ScalarTransform) = CompositeScalarTransform((t, s))
+Base.:∘(t::ScalarTransform, ct::CompositeScalarTransform) = CompositeScalarTransform((t, ct.transforms...))
+Base.:∘(ct::CompositeScalarTransform, t::ScalarTransform) = CompositeScalarTransform((ct.transforms..., t))
+Base.:∘(ct1::CompositeScalarTransform, ct2::CompositeScalarTransform) = CompositeScalarTransform((ct1.transforms..., ct2.transforms...))
+Base.:∘(t::ScalarTransform, tt::Vararg{ScalarTransform}) = CompositeScalarTransform((t, tt...))
+compose(args...) = CompositeScalarTransform(args)
```

Comment:
This is already achieved by CompositionsBase (https://github.com/JuliaFunctional/CompositionsBase.jl): It defines `compose` as an alias of `∘`, and hence ensures that packages that extend `∘` also share a common non-unicode alternative. Maybe use it instead of defining a different TransformVariables.compose?

Reply:
Seems like a good idea to me. My question then would be, should `compose` get exported?